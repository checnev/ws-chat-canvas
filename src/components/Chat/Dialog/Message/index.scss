.message {
  display: flex;
  flex-direction: column;
  margin-bottom: 15px;

  &__text {
    display: inline-block;
    padding: 8px;
    margin-bottom: 10px;
    border: 1px solid rgba(0, 0, 0, .1);
    border-radius: 4px;
    background:rgb(255, 255, 255);
    white-space: pre-line;
  }

  &__date {
    color: rgb(153, 153, 153);
    font-size: .8em;
  }

  &_direction {
    &_in {
      align-items: flex-start;
    }

    &_out {
      align-items: flex-end;

      .userInfo {
        display: none;
        justify-content: flex-end;

        &__avatar {
          order: 1;
        }

        &__username {
          margin-left: 0;
          margin-right: 10px;
        }
      }
    }
  }
}
