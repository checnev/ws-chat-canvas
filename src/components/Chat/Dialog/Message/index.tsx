import React from 'react';
import cn from 'utils/cn';
import UserInfo from 'components/UserInfo';
import type { IUser } from 'lib/user/User';

import './index.scss';

export interface IMessage {
  id: number;
  author: IUser;
  text: string;
  date: Date;
}

interface MessageProps {
  message: IMessage;
  direction: 'in' | 'out';
}

class Message extends React.Component<MessageProps> {
  render() {
    const message = cn('message');
    const { author, date, text } = this.props.message;
    return (
      <div
        className={
          message({ direction: this.props.direction })
        }
      >
        <div className={message('text')}>
          {text}
        </div>
        <div className={message('author')}>
          <UserInfo user={author} />
          <span className={message('date')}>
            {date.toLocaleString().replace(',', '')}
          </span>
        </div>
      </div>
    );
  }
}

export default Message;
