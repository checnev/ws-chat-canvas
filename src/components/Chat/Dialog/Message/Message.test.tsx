import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import Message from 'components/Chat/Dialog/Message';

let container: any = null;

beforeEach(() => {
  container = document.createElement('div');
  document.body.append(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

const user = {
  name: 'Naruto',
  color: 'green',
};

const message = {
  id: 1,
  author: user,
  text: 'Test Message',
  date: new Date(2021, 0, 30),
}

describe('Message component', () => {
  const setUp = (props: any) => <Message {...props} />;

  it('should render incoming Message', () => {
    act(() => {
      render(setUp({ message, direction: 'in' }), container);
    });

    expect(
      pretty(container.innerHTML)
    ).toMatchSnapshot();
  });

  it('should render outcoming Message', () => {
    act(() => {
      render(setUp({ message, direction: 'out' }), container);
    });

    expect(
      pretty(container.innerHTML)
    ).toMatchSnapshot();
  });
});
