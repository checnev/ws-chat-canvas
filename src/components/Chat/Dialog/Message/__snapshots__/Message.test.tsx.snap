// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component should render incoming Message 1`] = `
"<div class=\\"message message_direction_in\\">
  <div class=\\"message__text\\">Test Message</div>
  <div class=\\"message__author\\">
    <div class=\\"userInfo\\">
      <div class=\\"userInfo__avatar\\" style=\\"background-color: green;\\">N</div><span class=\\"userInfo__username\\">Naruto</span>
    </div><span class=\\"message__date\\">30.01.2021 00:00:00</span>
  </div>
</div>"
`;

exports[`Message component should render outcoming Message 1`] = `
"<div class=\\"message message_direction_out\\">
  <div class=\\"message__text\\">Test Message</div>
  <div class=\\"message__author\\">
    <div class=\\"userInfo\\">
      <div class=\\"userInfo__avatar\\" style=\\"background-color: green;\\">N</div><span class=\\"userInfo__username\\">Naruto</span>
    </div><span class=\\"message__date\\">30.01.2021 00:00:00</span>
  </div>
</div>"
`;
